abci/client/grpc_client.go

```diff
@@ -300,6 +301,45 @@ func (cli *grpcClient) ApplySnapshotChunkAsync(params types.RequestApplySnapshot
 	return cli.finishAsyncCall(req, &types.Response{Value: &types.Response_ApplySnapshotChunk{ApplySnapshotChunk: res}})
 }
 
+func (cli *grpcClient) PrepareProposalAsync(
+	params types.RequestPrepareProposal,
+) *ReqRes {
+
+	req := types.ToRequestPrepareProposal(params)
+	res, err := cli.client.PrepareProposal(context.Background(), req.GetPrepareProposal(), grpc.WaitForReady(true))
+	if err != nil {
+		return nil
```

Comment:
You might want to stop the grpc client here for consistency with other asynchronous ABCI calls:
```go
cli.StopForError(err)
```

Same applies to `ProcessProposalAsync` below